lib/Dialect/AIE/Transforms/AIEAssignBuffers.cpp

```diff
@@ -54,10 +54,11 @@ struct AIEAssignBufferAddressesPass
       }
     }
 
-    int max_data_memory_size =
-        (getTargetArch(m) == AIEArch::AIE2) ? 0x10000 : 0x8000;
 
     for (auto tile : m.getOps<TileOp>()) {
+      const auto &device_model = getDeviceModel(tile);
+      int max_data_memory_size =
+        (device_model.getTargetArch() == AIEArch::AIE2) ? 0x10000 : 0x8000;
```

Comment:
Should this be in the device model?
e.g. `device_model.getTargetArch().getMaxDataMemorySize()`